### PR TITLE
feat: add other examples and move hero into examples package

### PR DESCRIPTION
**Pull request by rube-de**
- move OPL-hero in examples folder
- integrate new examples 
  - OPL-Ping
  - Celer-Ping
  - Router PingPong

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,63 +1,9 @@
-# OPL Demo
+# Demo OPL
```
Comment (aefhm) [inline README.md:1]: I do go back and forth if we should just rename this to `oasis-cross-chain`.

File: README.md [hunk 2]
```diff
@@ -1,63 +1,9 @@
...
 
-This project demonstrates how to use [OPL] on Oasis Sapphire. Similar to this
-Celer [example] which passes messages between two smart contracts deployed on
-different chains. In this example, we will deploy one contract to Sapphire, and
-another to BSC.
+This repository includes several examples how to use cross-chain messaging
```
Comment (aefhm) [inline README.md]:
```suggestion
This repository includes several examples on how to use cross-chain messaging
```

File: examples/celer-ping/LICENSE [hunk 3]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (aefhm) [inline examples/celer-ping/LICENSE]: I think we might be synced on [Apache](https://github.com/oasisprotocol/demo-starter/blob/0a77e5ae0207178267e2d8ccc28dae9a337144a2/LICENSE).

File: examples/celer-ping/README.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
+# PingPong via Celer IM
+This example example lets you send a message via the Celer inter-chain messaging ([IM])
```
Comment (aefhm) [inline examples/celer-ping/README.md:3]: (nit)

```suggestion

This example example lets you send a message via the Celer inter-chain messaging ([IM])
```

File: examples/celer-ping/README.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+- Deploy contracts on both testnets (BSC Testent, Sapphire Testnet)
+- Send a message (`"Hello from BSC"`) from the BSC Testnet to the Sapphire Testnet
+- Listen to the messege sent via OPL on the Sapphire Testnet
+
+If you want to run the steps one at a time, read the following chapters.
+
+### Deploying Contracts
+
+Deploy the contracts on the host network (BSC Testnet) and the enclav network
```
Comment (aefhm) [inline examples/celer-ping/README.md]:
```suggestion
Deploy the contracts on the host network (BSC Testnet) and the enclave network
```

File: examples/celer-ping/contracts/IMessageBus.sol [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+interface IMessageBus {
+    function sendMessage(
+        address _receiver,
+        uint256 _dstChainId,
+        bytes calldata _message
+    ) external payable;
+
+    function calcFee(bytes calldata _message) external view returns (uint256);
+}
```
Comment (aefhm) [inline examples/celer-ping/contracts/IMessageBus.sol]:
```suggestion
}

```

File: examples/celer-ping/hardhat.config.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
+import { HardhatUserConfig } from "hardhat/config";
+import '@oasisprotocol/sapphire-hardhat';
+import "@nomicfoundation/hardhat-toolbox";
+import "hardhat-switch-network";
+// import "dotenv/config"
```
Comment (aefhm) [inline examples/celer-ping/hardhat.config.ts]:
```suggestion
```

File: examples/celer-ping/tasks/index.ts [hunk 8]
```diff
@@ -0,0 +1,160 @@
+import { task, subtask } from "hardhat/config";
+import { assert } from "console";
+
+task("full-pingpong")
```
Comment (aefhm) [inline examples/celer-ping/tasks/index.ts:4]: I like how pingpong denotes the direction of the message passing well.

**Review by aefhm — COMMENTED** [review]
Maybe it is time for us to borrow `demo-starter`'s linter too.

File: examples/celer-ping/LICENSE (hunk 3 above)
Comment (rube-de) [reply examples/celer-ping/LICENSE]: I see. Though the contract in the demo-starter has `// SPDX-License-Identifier: MIT`
So project apache and contracts MIT?

Comment (aefhm) [reply examples/celer-ping/LICENSE]: Great point. That does seem a little contradictory. I wonder if we just kept `MIT` to copy over from OpenZeppelin etc. cc @tjanez @matevz

File: examples/celer-ping/contracts/IMessageBus.sol [hunk 9]
```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: MIT
```
Comment (aefhm) [inline examples/celer-ping/contracts/IMessageBus.sol]: I'm seeing mixed info. https://github.com/oasisprotocol/dapp-blockvote/blob/28d57500f43b426854a58c044463ed4856deabbe/hardhat/contracts/test/TestToken.sol#L1

File: examples/celer-ping/tasks/index.ts [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+
+  task('verify-ping')
+  .addParam('pongAddr', 'Address of the Pong contract')
+  .addOptionalParam("message", "The message that should be bridged", "Hello from BSC")
+  .addOptionalParam("enclaveNetwork", "Network to deploy the Enclave contract on", "sapphire-testnet")
+  .setAction(async ({pongAddr, message, enclaveNetwork}, hre) => {
+    await hre.switchNetwork(enclaveNetwork);
+    console.log(`Verifying message on ${hre.network.name}...`);
+    let events;
+    const spinner = ['-', '\\', '|', '/'];
```
Comment (aefhm) [inline examples/celer-ping/tasks/index.ts:122]: Thank you for not using an NPM package. 🫶

File: examples/router-pingpong/scripts/pingpong.ts [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+
+async function main() {
+    const [owner] = await hre.ethers.getSigners();
+    const ownerAddress = await owner.getAddress();
+
+    // const gatewayAddress = "0x4b7ff2fae6b514a7943d360e01790dc1dfaf6736"; //arb-sepolia
+    // const gatewayAddress = "0x778a1f43459a05accd8b57007119f103c249f929"; //polygon-amoy
+
+    const pingPongAddress = "0x50F08578E4aD621BBDfEe8752832623928884658"; // arb-sepolia
+    // const pingPongAddress = "0x50F08578E4aD621BBDfEe8752832623928884658"; // polygon-amoy
```
Comment (aefhm) [inline examples/router-pingpong/scripts/pingpong.ts:14]: Maybe we want to store these somewhere.

File: examples/router-pingpong/scripts/pingpong.ts [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+    // pingPong.on(pingPong.filters.AckFromDestination, (requestId, ackMessage, event) => {
+    //     console.log(`AckFromDestination request id: ${requestId}, message: ${ackMessage}`);
+    //     event.removeListener();
+    // });
+    
+    // send ping
+    console.log("Sending ping...");
+    const newPing = await pingPong.iPing(destChaindId, destContractAddress, "hello ping", metadata);
+    const receipt = await newPing.wait();
+    console.log("Transaction hash: ", receipt.hash);
```
Comment (aefhm) [inline examples/router-pingpong/scripts/pingpong.ts:55]: Do we also wanna check the emitted event from the destination contract?

Comment (rube-de) [reply examples/router-pingpong/scripts/pingpong.ts:55]: yes, probably should rewrite the script to a task like I did with the others

Comment (rube-de) [reply examples/router-pingpong/scripts/pingpong.ts:55]: router protocol seems to have a problem on testnet (I pinged them already), so I can't verify my task script.

**Review by aefhm — APPROVED** [review]

File: examples/celer-ping/LICENSE (hunk 3 above)
Comment (rube-de) [reply examples/celer-ping/LICENSE]: changed contracts to Apache-2.0

**Pull request merged**